src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/CompletionRequestContextCacheTest.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Xunit;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
+{
+    public class CompletionRequestContextCacheTest
+    {
+        private Uri HostDocumentUri { get; } = new Uri("C:/path/to/file.razor");
+
+        private Uri ProjectedUri { get; } = new Uri("C:/path/to/file.foo");
+
+        private LanguageServerKind LanguageServerKind { get; } = LanguageServerKind.CSharp;
+
+        private CompletionRequestContextCache Cache { get; } = new CompletionRequestContextCache();
+
+        [Fact]
+        public void TryGet_SetRequestContext_ReturnsTrue()
+        {
+            // Arrange
+            var requestContext = new CompletionRequestContext(HostDocumentUri, ProjectedUri, LanguageServerKind);
+            var resultId = Cache.Set(requestContext);
+
+            // Act
+            var result = Cache.TryGet(resultId, out var retrievedRequestContext);
+
+            // Assert
+            Assert.True(result);
+            Assert.Same(requestContext, retrievedRequestContext);
+        }
+
+        [Fact]
+        public void TryGet_UnknownRequestContext_ReturnsTrue()
+        {
+            // Act
+            var result = Cache.TryGet(1234, out var retrievedRequestContext);
```

Comment:
Nit: a test that makes sure things don't return when the Cache actually has contents too would be good (even though chances are it'll pass/fail in lockstep with with this test).

Reply:
That's captured in the last test `TryGet_EvictedCompletionList_ReturnsFalse`